### ISSUE TITLE
Move / update from V3 to V4

### ISSUE THREAD

**Issue by barryboom** [donaldzou/WGDashboard#473]
**Describe The Problem**
i would like to move from v3 to V4 but unable to do so as currently no way to upgrade.

the inline upgrade breaks the dashboard "sudo ./wgd.sh update" from v3 to v4 end up breaking the configuration and features like backup / restore dont work.

i have tried may different ways and so far been unsuccesful so hoping for assistance from the greater minds assembled here.

what does work on both Ubuntu 22 / 24 is a clean install so what am hopipng collectively we can figure out is a way to 

1. have a fresh install of WG dash
2. the service to auto start after a server reboot
3. what files / config / package installs are required to be copied over from v3 to allow same WG clients operate on v4

in order to achieve this i am happy to build a new server, fresh install and use winscp to copy over files / databases as required.

**To Reproduce**
try and upgrade from V3 to V4 on ubuntu 22 or 24

**Comment by barryboom** [comment]
2. the service to auto start after a server reboot

This is no longer an issue found the autostart section and have managed to get that working when i do a fresh install of v4.1 on ubuntu 24.04.1

got a sperate server running dashboard V3.1 so happy to test out moving some files over from that to new server with V4 if anyone is able to advise on what files should be needed.

**Comment by DaanSelen** [comment]
You mean: the login breaks from moving from V3 to V4, right?

**Comment by DaanSelen** [comment]
> 2. the service to auto start after a server reboot
> 
> This is no longer an issue found the autostart section and have managed to get that working when i do a fresh install of v4.1 on ubuntu 24.04.1
> 
> got a sperate server running dashboard V3.1 so happy to test out moving some files over from that to new server with V4 if anyone is able to advise on what files should be needed.

This can be done by enabling the systemd service file. `systemctl enable <name of the dashboard>`

**Comment by barryboom** [comment]
> You mean: the login breaks from moving from V3 to V4, right?

no that can be fixed but other thing dont seem to work even after that like the backup / restore functions just log you out and cannot run debug as get a VENV error when trying to do any command like sudo wg.sh start/stop/debug even though oddly dashboard is still up and running.

**Comment by barryboom** [comment]
> > 2. the service to auto start after a server reboot
> > 
> > This is no longer an issue found the autostart section and have managed to get that working when i do a fresh install of v4.1 on ubuntu 24.04.1
> > got a sperate server running dashboard V3.1 so happy to test out moving some files over from that to new server with V4 if anyone is able to advise on what files should be needed.
> 
> This can be done by enabling the systemd service file. `systemctl enable <name of the dashboard>`

correct this is no longer an issue and autostart instruction do work

**Comment by DaanSelen** [comment]
Yeah I think the logging out when clicking backup is a known bug, not related to you.

Regarding the venv in debug mode? Can you post some screenshots or tell me how to reproduce that?

**Comment by barryboom** [comment]
> Yeah I think the logging out when clicking backup is a known bug, not related to you.
> 
> Regarding the venv in debug mode? Can you post some screenshots or tell me how to reproduce that?

i only get issues when i do an upgrade with the logging out. If a clean install no issues.

in order to reproduce 

1. ubuntu server 22.04 or 24.04
2. install WGdashboard v3xx
3. perform inline upgrade (sudo ./wgd.sh update)
4. delete password line in dashboard.ini file to fix logon issue
5. SSH conneciton to server CD to directory and try running any command like sudo .wgd.sh start or debug 

i will have perform these later aswell myself and get the error but for now going to try copying files from my v3 to my working v4 instance and see what happens there.

**Comment by DaanSelen** [comment]
I'll try to recreate it! Hope to come back soon.

**Comment by donaldzou** [comment]
Hi all, this is still persisting?

**Comment by barryboom** [comment]
yes with my limited knowledge been unable to succesfully migrate / upgrade from V3 to V4 and still welcome any suggestions.

i am happy to setup a new server and copy files over but always seems to run in to some issue.

**Comment by donaldzou** [comment]
> yes with my limited knowledge been unable to succesfully migrate / upgrade from V3 to V4 and still welcome any suggestions.
> 
> i am happy to setup a new server and copy files over but always seems to run in to some issue.

Hi @barryboom , sorry this is happening.. If you wish, you could create a brand new server and send me the credentials, and I can look into what's happening :)

**Comment by barryboom** [comment]
do you have a server you can try to migrate too i am happy to clone one of the servers running V3. i can succesfully manage that taking a server running v3.0.6.2 or even v3.1 (athough thats a little harder as unable to install from Github anymore) and migrate that to an new server but need to stick with same WGDAshboard version as cannot get V4 to work properly when ever i have tried copying things over like the db and files etc that i think would be needed.

**Comment by donaldzou** [comment]
Hi! So on my side I've done this and it is working, here are the commands I ran:

```bash
git clone https://github.com/donaldzou/WGDashboard -b v3.0.6.2 WGDashboard_v3

# Since I can't install package to global Python due to Ubuntu 22.02+ system policy, I'll need to install virtual environment
cd WGDashboard_v3/src
python3 -m venv ./venv

# Install packages to the virtual environment
./venv/bin/python3 -m pip install -r requirements.txt

# Run WGDashboard directly, this should work with no problem
./venv/bin/python3 dashboard.py

# Use control + c to terminate, and now we update to v4
./wgd.sh update

# Once done, we run install
./wgd.sh install

# Remove the password line from wg-dashboard.ini due to known issue
nano wg-dashboard.ini

# Once done, we run
./wgd.sh debug
```

Everything is fine to me, maybe it was due to you didn't run `./wgd.sh install` after update?

**Comment by barryboom** [comment]
OK i will have a check and run these commands but what i was hoping for was what can be done to move a V3 dashboard to a new V4 dashboard what files need copied over etc but this should help if it works.

**Comment by barryboom** [comment]
Hit a few issues so think a few steps are missing / different here

1. needed to install PIP 
2. unable to start port already in use( i have autostrart config)
3. changed port to alternative and kind of worked but when trying to upgrade get root@oldOVHshared:~/WGDashboard_v3/src# ./wgd.sh update
-bash: ./wgd.sh: Permission denied

if you want i can setup the server again and send over logon details for you to check my setup and what i am working with? its just a test server this one anyway so no issues if fails and needs wiped.

**Comment by barryboom** [comment]
ok tried now 4 times and still these instrucitons don't work get to the update section and give permisisons error just incase anyone else is trying to perform the same actions 

i have attached here the SSH process which i followed with the commands and outputs as i went through it

[Log.txt](https://github.com/user-attachments/files/18989603/Log.txt)

**Comment by DaanSelen** [comment]
Is this still an issue?

**Comment by barryboom** [comment]
Yes still not possible and no provided solution or workaround

**Comment by DaanSelen** [comment]
> ok tried now 4 times and still these instrucitons don't work get to the update section and give permisisons error just incase anyone else is trying to perform the same actions
> 
> i have attached here the SSH process which i followed with the commands and outputs as i went through it
> 
> [Log.txt](https://github.com/user-attachments/files/18989603/Log.txt)

Why are you deploying the old branch? Updating manually can probably be done through `git pull`

**Comment by barryboom** [comment]
> > ok tried now 4 times and still these instrucitons don't work get to the update section and give permisisons error just incase anyone else is trying to perform the same actions
> > i have attached here the SSH process which i followed with the commands and outputs as i went through it
> > [Log.txt](https://github.com/user-attachments/files/18989603/Log.txt)
> 
> Why are you deploying the old branch? Updating manually can probably be done through `git pull`

Because it is the old branch that i have installed on multiple servers which i hope to be able to migrate from v3 to v4 but currently seems impossible. So to test and not effect prodution i need to create copies of PROD servers to perform the testing on.

**Comment by barryboom** [comment]
@donaldzou i assume you never got round to testing out the migration with the test server details i sent over ?

**Comment by donaldzou** [comment]
@barryboom yes please, could you also let me know on GitHub once you sent it? Sometimes it will be in my junk folder lol

**Comment by barryboom** [comment]
> [@barryboom](https://github.com/barryboom) yes please, could you also let me know on GitHub once you sent it? Sometimes it will be in my junk folder lol

new details sent over. Setup a new test server same as the several others that ideally can be upgrade by hook or by crook from V3 to V4

**Comment by barryboom** [comment]
Delighted to say this is now resolved and there is a possible path to upgrade

**Comment by chop249** [comment]
> Delighted to say this is now resolved and there is a possible path to upgrade 

@barryboom could you give us the solution? I have the same issue only on Debian. Thanks.
